pkg/notifications/notification_test.go

```diff
@@ -0,0 +1,83 @@
+// SPDX-FileCopyrightText: 2024 Greenbone AG <https://greenbone.net>
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+package notifications
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"sync/atomic"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestClient_CreateNotification(t *testing.T) {
+	tests := []struct {
+		name                   string
+		serverHttpResponseCode int
+		wantErr                bool
+	}{
+		{
+			name:                   "notification can be sent",
+			serverHttpResponseCode: http.StatusOK,
+			wantErr:                false,
+		},
+		{
+			name:                   "client returns error on notification service error",
+			serverHttpResponseCode: http.StatusInternalServerError,
+			wantErr:                true,
+		},
+	}
+
+	// client arguments
+	notification := Notification{
+		Title: "test notification",
+		// other mandatory fields not set for brevity ...
+	}
+
+	wantNotificationSerialized, err := json.Marshal(notification)
```

Comment:
Just a matter of taste: you could think about proving the notification as explicit json string. It's definitely not necessary, but I always like to see what the json objects "really" look like :-)